File: src/main/java/trainticket/Tester.java
package trainticket;

import java.util.Scanner;

public class Tester {

    public static void main (String[] args) {
        startBuyTicket();
    }

    private static void startBuyTicket() {

        int ticketNum;
        int roundTripNum;

        System.out.println("Please enter number of tickets");
        Scanner scanner = new Scanner(System.in);
        ticketNum = scanner.nextInt();

        if(checkTicketNum(ticketNum)) {

            System.out.println("How many round-trip tickets");
            roundTripNum = scanner.nextInt();

            Ticket ticket = new Ticket(ticketNum, roundTripNum);
            if(roundTripNum > 0 && roundTripNum <= ticketNum) {
                ticket.getTotal();
                startBuyTicket();
            } else {
                System.out.println("Round-trip tickets can not over than total ticket.");
                startBuyTicket();
            }

        } else {
            endBuyTicket();
        }

    }

    private static void endBuyTicket() {
        System.out.println("Finish Buy Ticket");
        System.exit(0);
    }

    private static boolean checkTicketNum(int ticketNum) {

//        if(ticketNum < 0)
//            System.out.println("Ticket Number must bigger then 0.");

        return ticketNum > 0;
    }

}

class Ticket {

    private int ticketNum;
    private int roundTripNum;
    private int singleTripNum;

    Ticket(int ticketNum, int roundTripNum) {

        this.ticketNum = ticketNum;
        this.roundTripNum = roundTripNum;
        singleTripNum = ticketNum - roundTripNum;
    }

    int getTicketNum() {
        return ticketNum;
    }

    int getRoundTripNum() {
        return roundTripNum;
    }

    void getTotal() {
        double single = 1000;
        double roundTrip = 2000 * 0.9;
        double total = 0;
        System.out.println("Total tickets: " + ticketNum);
        System.out.println("Round-trip: " + roundTripNum);
        total = (singleTripNum * single) + (roundTripNum * roundTrip);
        System.out.println("Total: " + total);
    }
}

File: src/main/kotlin/trainticket/Tester.kt
package trainticket

import java.text.DecimalFormat
import java.util.*
import kotlin.system.exitProcess

fun main(args : Array<String>) {
    startBuyTicket()
}

fun startBuyTicket() {

    var ticketNum: Int
    var roundTripNum: Int
    var scanner = Scanner(System.`in`)
    println("Please enter number of tickets")
    ticketNum = scanner.nextInt()

    if(ticketNum >= 0) {

        println("How many round-trip tickets : ")

        roundTripNum = scanner.nextInt()

        if(ticketNum > 0 && roundTripNum <= ticketNum) {
//            val ticket = BuyTicket(ticketNum, roundTripNum)
//            ticket.printTotalFee()
            BuyTicket(ticketNum, roundTripNum).printTotalFee()
            startBuyTicket()
        } else {
            println("You don't buy any ticket or round trip ticket is more than total ticket. \n Please try again.")
            startBuyTicket()
        }

    } else {
        exitProcess(0)
    }

}

class BuyTicket (private val totalTicket : Int, private val roundTripTicket : Int) {

    // 相當於Java中 類裡面的static變量
    companion object{
        var singleFee = 1000
        var roundTripTicketFee = 2000 * 0.9
    }

//    fun printTotalFee() {
//
//        println("Total Ticket: $totalTicket\nRound-trip: $roundTripTicket\nTotal: $totalPrice")
//
//    }

    private val totalPrice = DecimalFormat("0").format((totalTicket - roundTripTicket) * singleFee + roundTripTicket * roundTripTicketFee)
    fun printTotalFee() = println("Total Ticket: $totalTicket\nRound-trip: $roundTripTicket\nTotal: $totalPrice")

}
